jerry-core/ecma/operations/ecma-function-object.c

```diff
@@ -802,7 +802,20 @@ ecma_op_function_get_function_realm (ecma_object_t *func_obj_p) /**< function ob
     const ecma_compiled_code_t *bytecode_data_p = ecma_op_function_get_compiled_code (ext_function_obj_p);
     return ecma_op_function_get_realm (bytecode_data_p);
   }
-
+#if ENABLED (JERRY_BUILTIN_PROXY)
+  if (ECMA_OBJECT_IS_PROXY (func_obj_p))
+  {
+    ecma_proxy_object_t *proxy_obj_p = (ecma_proxy_object_t *) func_obj_p;
+    if (ecma_is_value_null (proxy_obj_p->handler))
+    {
+      ecma_raise_type_error (ECMA_ERR_MSG ("Prototype from revoked Proxy is invalid."));
+      return NULL;
+    }
+    ecma_extended_object_t *ext_obj_p = (ecma_extended_object_t *) ecma_get_object_from_value (proxy_obj_p->target);
+    const ecma_compiled_code_t *bytecode_data_p = ecma_op_function_get_compiled_code (ext_obj_p);
```

Comment:
What happens if the proxy object's target is an other proxy object?

You should return with the `ecma_op_function_get_function_realm (proxy->target)`.

This should probably fix the `<test id="built-ins/Proxy/get-fn-realm-recursive.js"><reason></reason></test>` as well.

Reply:
Good idea !